src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -176,6 +213,10 @@ public FileStream(string path, FileMode mode, FileAccess access, FileShare share
             {
                 throw new ArgumentOutOfRangeException(nameof(bufferSize), SR.ArgumentOutOfRange_NeedPosNum);
             }
+            else if (preallocationSize < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(preallocationSize), SR.ArgumentOutOfRange_NeedNonNegNum);
+            }
```

Comment:
I do prefer that you throw on negatives but I wonder why did you choose to switch? 
Also, the PR description says the following:
>Value provided as allocationSize argument has no effect unless it's positive and a regular file is being created, overwritten, or replaced. It means that:
>
>* if value is <= 0 it's just ignored.
>* if the path points to a different type of file like pipe etc the setting is also just ignored
>* if the file system does not support such operation, the setting is also ignored

With the exception of bullet 1 (which clearly no longer applies) bullets 2 & 3 still apply?

Reply:
> why did you choose to switch?

@carlossanlop expressed his opinion that it was a bad idea, so I've changed it.

@Jozkee Based on your feedback I've updated the xml docs and I am also going to update the PR description